handler/src/main/java/io/netty/handler/traffic/GlobalChannelTrafficShapingHandler.java

```diff
@@ -299,15 +300,13 @@ public void setMaxDeviation(float maxDeviation, float slowDownFactor, float acce
         if (maxDeviation > MAX_DEVIATION) {
             throw new IllegalArgumentException("maxDeviation must be <= " + MAX_DEVIATION);
         }
-        if (slowDownFactor < 0) {
-            throw new IllegalArgumentException("slowDownFactor must be >= 0");
-        }
+
         if (accelerationFactor > 0) {
             throw new IllegalArgumentException("accelerationFactor must be <= 0");
         }
         this.maxDeviation = maxDeviation;
         this.accelerationFactor = 1 + accelerationFactor;
-        this.slowDownFactor = 1 + slowDownFactor;
+        this.slowDownFactor = 1 + checkPositive(slowDownFactor, "slowDownFactor");
```

Comment:
this change makes the behaviour different to what we have now.. as you moved the positive check here we may already upped some fields before throwing. Please move the check before assignments